fastly/errors.go

```diff
@@ -318,12 +324,31 @@ func NewHTTPError(resp *http.Response) *HTTPError {
 		return &e
 	}
 
-	// If this is a jsonapi response, decode it accordingly
-	if resp.Header.Get("Content-Type") == jsonapi.MediaType {
+	switch resp.Header.Get("Content-Type") {
+	case jsonapi.MediaType:
+		// If this is a jsonapi response, decode it accordingly
 		if err := decodeBodyMap(resp.Body, &e); err != nil {
 			panic(err)
 		}
-	} else {
+
+	case "application/problem+json":
+		// Response is a "problem detail" as defined in RFC 7807.
+		var problemDetail struct {
+			URL    string `json:"type,omitempty"`   // URL to a human-readable document describing this specific error condition
+			Title  string `json:"title,omitempty"`  // A short name for the error type, which remains constant from occurrence to occurrence
+			Status int    `json:"status"`           // HTTP status code
+			Detail string `json:"detail,omitempty"` // A human-readable description of the specific error, aiming to help the user correct the problem
+		}
+		if err := json.NewDecoder(resp.Body).Decode(&problemDetail); err != nil {
+			panic(err)
```

Comment:
Yes, anything other than `panic` seems like it'd be an improvement here. I'd like to make that change in a separate PR since it'll involve modifying the (unrelated) original `panic`.